frog/imports/client/Wiki/ModalPageSettings.js

```diff
@@ -201,64 +208,101 @@ class NewPageModal extends React.Component<PropsT, StateT> {
                 }}
                 data-testid="wiki_page_title_editor"
               />
-              {errorDiv !== null && (
-                <FormHelperText error>{errorDiv}</FormHelperText>
+              {this.state.error !== null && (
+                <FormHelperText error>{error}</FormHelperText>
               )}
             </FormControl>
           </FormGroup>
-          <Collapse in={expanded} timeout="auto">
-            <AppBar position="static">
-              <Tabs
-                value={this.state.currentTab}
-                indicatorColor="secondary"
-                onChange={this.handleTabs}
-                variant="fullWidth"
-              >
-                <Tab label="Settings" className={classes.tabs} />
-                <Tab label="Component" className={classes.tabs} />
-                <Tab label="Operator" className={classes.tabs} />
-              </Tabs>
-            </AppBar>
+          <Collapse
+            in={expanded || this.props.action === 'edit'}
+            timeout="auto"
+          >
+            {this.props.action === 'create' && (
+              <AppBar position="static">
+                <Tabs
+                  value={this.state.currentTab}
+                  indicatorColor="secondary"
+                  onChange={this.handleTabs}
+                  variant="fullWidth"
+                >
+                  <Tab label="Settings" className={classes.tabs} />
+                  <Tab label="Component" className={classes.tabs} />
+                  <Tab label="Operator" className={classes.tabs} />
+                </Tabs>
+              </AppBar>
+            )}
             <DialogContent classes={{ root: classes.modalInner }}>
               {currentTab === 0 && (
-                <FormGroup>
-                  <FormControl>
-                    <Typography variant="h6">Social Plane</Typography>
-                    <Select
-                      value={this.state.socialPlane}
-                      onChange={this.handleSocialPlaneChange}
-                      id="social-plane"
-                      className={classes.selectSocialPlane}
-                    >
-                      <MenuItem value={3}>Everyone</MenuItem>
-                      <MenuItem value={1}>Each Individual</MenuItem>
-                    </Select>
-                  </FormControl>
-                  <FormGroup row>
-                    <FormControlLabel
-                      control={
-                        <Checkbox
-                          checked={this.state.allowView}
-                          onChange={this.handleChangeAllowView}
-                          color="primary"
+                <>
+                  <FormGroup>
```

Comment:
We should consider using react-jsonschema-form as this view logic is quite complex